### ISSUE TITLE
Should we try to repair crashed database tables?

### ISSUE THREAD

**Issue by willmot** [humanmade/backupwordpress#836]
Crashed db tables cause the backup to fail. (https://github.com/humanmade/backupwordpress/issues/811)

Should we:

- Attempt to better detect and show the user a warning message along with a link to the Codex article on how they can use WordPress to try to repair the table themselves.
- Just attempt to repair the table ourselves automatically.

**Comment by owaincuvelier** [comment]
Also see: https://github.com/WordPress/WordPress/blob/master/wp-admin/maint/repair.php

**Comment by willmot** [comment]
Also see: https://codex.wordpress.org/Editing_wp-config.php#Automatic_Database_Optimizing

**Comment by owaincuvelier** [comment]
Maybe run a DB Repair before each SQL Dump? Just cuz'

**Comment by willmot** [comment]
Yeah would be good to research what the downsides of doing that are. I assume there is a reason that Core doesn't just do that?

**Comment by owaincuvelier** [comment]
I can't imagine a reason why core would. But we're doing a backup, and it does make sense to check everything's in order before taking the backup. What if the backup works but we've backed up a semi-corrupt db or something. It was just a thought regardless :)
